### ISSUE TITLE
Carousel "Uncaught type error: cannot read property persist of undefined"

### ISSUE THREAD

**Issue by sschenk** [react-bootstrap/react-bootstrap#2029]
I'm trying to implement a controlled carousel  with react redux but I'm getting the error in subject. 
Following a stack overflow question I've posted, there might be a bug in the react-bootstrap code.

Here is the stack overflow question: http://stackoverflow.com/questions/38222883/uncaught-type-error-cannot-read-property-persist-of-undefined-with-react-bootst?noredirect=1#comment63874022_38222883

and here is my code

Container:

```
import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router'
import store from 'store/configureStore'
import Slides from 'components/SlideShow'
import { getInitalSlides, handleSelect } from 'actions/SlidesActions'

class Home extends Component {

constructor(props) {
    super(props)
    this.state = {
        index: null,
        direction: null
    }
    this.handleSelect = this.handleSelect.bind(this)    

static fetchData({ store }) {
        return store.dispatch(getInitalSlides())
    }

    componentDidMount() {
        this.props.getInitalSlides()
    }

handleSelect(selectedIndex, e) {
        //alert(e)
        this.props.handleSelect(selectedIndex, e)
    }
```
```

  render() {
    return (
      <div className="Home">
        <h1>Home Page</h1>
        <Slides 
        slides={this.props.slides}   
        getInitialState={this.state.index} 
        getInitialStateD={this.state.direction}
        slidesControl={this.handleSelect}
        />
        <div><Link to="/question">to question</Link></div>
        <div><Link to="/posts">to posts</Link></div>
      </div>
    )
  }
}

function mapStateToProps (state) {
  const { slides, handleSelect } = state

  return { slides: state.slides, onSelect: state.handleSelect } 
}

export { Home }
export default connect(mapStateToProps { getInitalSlides, handleSelect})(Home)
```

Component:

   ```
 render() {

    return (
      <Carousel 
      activeIndex={this.props.getInitialState} 
      direction={this.props.getInitialStateD} 
      onSelect={(selectedIndex,e)=>this.props.slidesControl(selectedIndex,e)}
      >
      {

      this.props.slides.map((s)=>{
              let id = s.get('id')
              let title = s.get('title')
              let image = s.get('image')
              let alt = s.get('alt')
              let caption = s.get('caption')
              return(

                    <Carousel.Item key={id} >
                    <img width={900} height={500} alt={s.get('alt')} src={image} alt={alt}/>
                    <Carousel.Caption>
                      <h3>{title}</h3>
                      <p>{caption}</p>
                    </Carousel.Caption>
                  </Carousel.Item>

              )
      }) 



    }
    </Carousel>)
  }
}
```

action creator:

```
export const HANDLE_SELECT = Symbol('HANDLE_SELECT')
export function handleSelect(selectedIndex, e) {
	//alert('selected=' + selectedIndex + ', direction=' + e.direction)
    
    return {
    	index: selectedIndex,
    	direction: e.direction,
    	type: HANDLE_SELECT
    }
}
```

**Comment by lkrnac** [comment]
I did quick analysis of this problem: http://stackoverflow.com/a/38226823/1919879

**Comment by taion** [comment]
I have a fix pending.

**Comment by sschenk** [comment]
Great, when do you think it'll be ready to ship? Do you need help testing etc.?

**Comment by taion** [comment]
This is resolved in #2036
